app/views/hyrax/file_sets/_show_actions.html.erb

```diff
@@ -3,7 +3,8 @@
     <%= link_to t('.analytics'), @presenter.stats_path, id: 'stats', class: 'btn btn-default' %>
   <% end %>
 
-  <% if @presenter.editor? %>
+  <% no_workflow_restrictions = !( @presenter.respond_to?( :workflow_restriction? ) and @presenter.workflow_restriction? ) %>
```

Comment:
In what cases does `@presenter` not respond to `#workflow_restriction?`? Would it be possible to just ensure that all appropriate presenters have this method?

Alternatively `<% if @presenter.editor? && !@presenter.try(:workflow_restriction?) %>` could be easier to follow.

Do we need a unit test for the new case in https://github.com/samvera/hyrax/pull/3881/files#diff-c82771edaa1f35232e9c45b621988f7c?

Reply:
Oh, i'm just realizing how many times this `no_workflow_restrictions` code is being repeated. Let's definitely figure out how to move this into the presenters. Duplicating this code in many views is going to be very fragile, I fear.

Reply:
I've updated the code to rely on a method on the presenters, but not all presenters have this method, and there are a number of spec tests that don't bother with presenters and I chose not track them all down and fix and instead used a try on the method call.